tests/test_mtp.py

```diff
@@ -0,0 +1,226 @@
+import typing
+
+import pytest
+import torch
+
+from fast_llm.config import UpdateType
+from fast_llm.engine.distributed.config import DistributedConfig
+from fast_llm.engine.distributed.distributed import Distributed
+from fast_llm.layers.language_model.config import LanguageModelKwargs, LanguageModelLossNames
+from fast_llm.layers.language_model.embedding import WORD_EMBEDDINGS_WEIGHT
+from fast_llm.layers.language_model.head import OUTPUT_WEIGHTS
+from fast_llm.layers.transformer.config import TransformerConfig, TransformerKwargs
+from fast_llm.models.gpt.config import GPTBaseModelConfig
+from fast_llm.models.gpt.model import GPTBaseModel
+from fast_llm.utils import Assert
+from tests.common import requires_cuda
+
+try:
+    from fast_llm.layers.ssm.config import SSMConfig
+    from fast_llm.layers.ssm.discrete_mamba2 import DiscreteMamba2
+    from fast_llm.layers.ssm.mamba_layer import MambaLayer
+    from fast_llm.models.ssm.model import HybridSSMBaseModel, HybridSSMBaseModelConfig
+except ImportError:
+    MambaLayer, HybridSSMBaseModel, HybridSSMBaseModelConfig, DiscreteMamba2 = (
+        None,
+        None,
+        None,
+        None,
+    )
+    # Mamba not installed, skipping tests
+
+
+run_hybrid_test = MambaLayer is not None and DiscreteMamba2 is not None and torch.cuda.is_available()
+
+
+SEQUENCE_LENGTH = 200
+BATCH_SIZE = 4
+HIDDEN_SIZE = 256
+VOCAB_SIZE = 500
+
+
+def materialize_meta_tensors(model, tensor_space):
```

Comment:
Duplicate from `test_ssms`? Let's move to a util file (`common.py`?) and import.